src/metabase/driver/mysql.clj

```diff
@@ -86,6 +101,46 @@
 (defn- date-format [format-str expr] (hsql/call :date_format expr (hx/literal format-str)))
 (defn- str-to-date [format-str expr] (hsql/call :str_to_date expr (hx/literal format-str)))
 
+(defn- timezone-id->offset-str
+  "Get an appropriate timezone offset string for a timezone with `timezone-id`. MySQL only accepts these offsets as
+  strings like `-8:00`.
+
+      (timezone-id->offset-str \"US/Pacific\") ; -> \"-08:00\"
+
+  Returns `nil` if `timezone-id` is itself `nil`."
+  [^String timezone-id]
+  (when timezone-id
+    (time/unparse (time/formatter "ZZ" (t/time-zone-for-id timezone-id)) (t/now))))
```

Comment:
It's a good idea to `def` the formatter so it doesn't re-parse that format string and recreate the formatter objects on each call. Unlike the java.util stuff, the JodaTime formatters are immutable and threadsafe.

Reply:
I'm switching this to `def` the formatter but since we need different timezones which are subject to change I still have to use [`.withZone`](http://www.joda.org/joda-time/apidocs/org/joda/time/format/DateTimeFormatter.html#withZone-org.joda.time.DateTimeZone-) to get a new formatter with the correct timezone each time. So we're still saving some effort by avoiding repeated parsing but we will still have to allocate new objects. Luckily our MBQL queries rarely have more than a handful of timestamps in them so that is probably the least of our worries